motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -1968,14 +1968,29 @@ int Ros_MotionServer_GetDhParameters(Controller* controller, SimpleMsg* replyMsg
 
 int Ros_MotionServer_SetSelectedTool(Controller* controller, SimpleMsg* receiveMsg, SimpleMsg* replyMsg)
 {
+#if !(DX100 || FS100)

```

Comment:
hm, so this doesn't work on DX100 nor FS100?

Reply:
This is a new-ish API that apparently wasn't back-ported to previous controllers. (It's not included in the header files anyway.)

Reply:
In `v1.3.0` it's in `inc/mpMotCtrl.h`:

```c
extern int mpMotSetTool(int grpNo, int toolNo);
```
I believe?

Reply:
moMotSetTool is used for a different motion API. The correct function is `mpSetToolNo`. 
However, I did check this again. Turns out that it is available for DX100, but is not available for FS100. I have updated this in a new commit.

Reply:
Oh right, apologies. I confused the two. Part of the legacy API for newer controllers it seems.